core/src/main/java/org/apache/calcite/sql/fun/SqlJsonExistsFunction.java

```diff
@@ -34,14 +34,15 @@ public SqlJsonExistsFunction() {
     super("JSON_EXISTS", SqlKind.OTHER_FUNCTION,
         ReturnTypes.cascade(ReturnTypes.BOOLEAN, SqlTypeTransforms.FORCE_NULLABLE), null,
         OperandTypes.or(
-            OperandTypes.family(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER),
```

Comment:
I think the main reason these JSON functions define the first operand as sql type family `ANY` is that the first operand usually is with struct(or JSON type), calcite doesn't support JSON type yet, use ANY will translate the operand as a Java `Object`.

Reply:
If calcite has supported JSON type, it can add some  `OperandTypes.family` to SqlOperandTypeChecker. Using SqlTypeFamily.ANY will prevent type validation.

Reply:
Calcite doesn't support JSON type, so `ANY` is the only choice.

Reply:
Right. In our code base JSON functions usually accepts both character string and JSON inputs. We don't have a individual data type for JSON so we use `ANY` instead. See a ticket of JSON data type support:

https://issues.apache.org/jira/browse/CALCITE-2869